src/finch/tensor.py

```diff
@@ -156,12 +159,25 @@ def __mod__(self, other):
     def __pow__(self, other):
         return self._elemwise_op("^", other)
 
-    def __matmul__(self, other: "Tensor") -> "Tensor":
-        if self.ndim < 2 or other.ndim < 2:
-            raise NotImplementedError(
-                "`@` is not implemented for arrays with less than two dimensions."
+    @compiled()
+    def __matmul__(self, other: Tensor) -> Tensor:
```

Comment:
Can we do it the other way around?
Instead of using `complied()` here, can we import `from .compiled import lazy, compute` inside `def __matmul__` and wrap `self` and `other`? This way we can revert `compiled.py` file changes completely.

Reply:
There are a number of issues with this approach:
- It won't get past the recursive import since we're still doing `from .compiled import ...`
- It will directly compute the matmul, even if it is part of a lazy evaluation, which is what we'd like to avoid.

Reply:
Sure! Thanks for an explanation!